src/Domain.LinnApps/Reports/StoragePlaceAuditReportService.cs

```diff
@@ -0,0 +1,151 @@
+﻿namespace Linn.Stores2.Domain.LinnApps.Reports
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Stores2.Domain.LinnApps.Stock;
+
+    public class StoragePlaceAuditReportService : IStoragePlaceAuditReportService
+    {
+        private readonly IReportingHelper reportingHelper;
+
+        private readonly IRepository<StockLocator, int> stockLocatorRepository;
+
+        public StoragePlaceAuditReportService(
+            IReportingHelper reportingHelper,
+            IRepository<StockLocator, int> stockLocatorRepository)
+        {
+            this.reportingHelper = reportingHelper;
+            this.stockLocatorRepository = stockLocatorRepository;
+        }
+
+        public ResultsModel StoragePlaceAuditReport(IEnumerable<string> locationList, string locationRange)
+        {
+            var stockLocators = new List<StockLocator>();
+
+            if (!string.IsNullOrEmpty(locationRange) && locationRange.StartsWith("E-K"))
+            {
+                stockLocators = this.stockLocatorRepository
+                    .FilterBy(s => s.CurrentStock == "Y" && s.StorageLocation.LocationCode.StartsWith(locationRange))
+                    .OrderBy(s => s.StorageLocation.LocationCode).ThenBy(s => s.PartNumber).ToList();
```

Comment:
you could make this method async and then await a .ToListAsync if you wanted to buy into my async crusade

Reply:
i knew you'd say that! will do

Reply:
lol - like I said it will probably require a bit of a tests rewrite so not sure its worth it. can revisit at the next show & tell and maybe even use this one as an example we could do together if you decide not to go async

Reply:
ok sounds like a good plan. it would be good for everyone including me to see the different repo testing in action